File: benchmarks/algorithm/sort.go
package main

import (
    "fmt"
    "time"
)

func min3(a, b, c int) int {
    if a <= b {
        if a <= c { return a } else { return c }
    } else {
        if b <= c { return b } else { return c }
    }
}

func max3(a, b, c int) int {
    if a >= b {
        if a >= c { return a } else { return c }
    } else {
        if b >= c { return b } else { return c }
    }
}

func mid3(a, b, c int) int {
    return a + b + c - min3(a, b, c) - max3(a, b, c)
}

func main() {
    a, b, c := 9, 3, 7
    start := time.Now()
    lo := min3(a, b, c)
    mi := mid3(a, b, c)
    hi := max3(a, b, c)
    elapsed := time.Since(start).Nanoseconds()
    fmt.Printf("%d %d %d (%dns)\n", lo, mi, hi, elapsed)
}

File: benchmarks/algorithm/sort.rs
use std::time::Instant;

fn min3(a: i32, b: i32, c: i32) -> i32 {
    if a <= b { if a <= c { a } else { c } }
    else { if b <= c { b } else { c } }
}

fn max3(a: i32, b: i32, c: i32) -> i32 {
    if a >= b { if a >= c { a } else { c } }
    else { if b >= c { b } else { c } }
}

fn mid3(a: i32, b: i32, c: i32) -> i32 {
    a + b + c - min3(a, b, c) - max3(a, b, c)
}

fn main() {
    let a = 9; let b = 3; let c = 7;
    let start = Instant::now();
    let lo = min3(a, b, c);
    let mi = mid3(a, b, c);
    let hi = max3(a, b, c);
    let elapsed = start.elapsed().as_nanos();
    println!("{} {} {} ({}ns)", lo, mi, hi, elapsed);
}
